assets/src/scripts/monitoring-location/components/hydrograph/selectors/discrete-data.js

```diff
@@ -20,29 +42,38 @@ export const getGroundwaterLevelPoints = createSelector(
             return [];
         }
         return gwLevels.values.map(data => {
+            const approvalDetails = getDetailsForApprovalCode(data.qualifiers);
             return {
                 value: data.value,
-                dateTime: data.dateTime
+                dateTime: data.dateTime,
+                classes: [GW_LEVEL_CLASS, approvalDetails.class],
+                label: approvalDetails.label,
+                radius: GW_LEVEL_RADIUS
             };
         });
     }
 );
 
 /*
-* When given an approval code, will return the text equivalent
-*  @param {String} approvalCode - Usually a letter such as 'A'
-*   @return {String} - an easy to understand text version of an approval code
-*/
-const approvalCodeText = function(approvalCode) {
-    const approvalText = {
-        P: 'Provisional',
-        A: 'Approved',
-        R: 'Revised',
-        default: `unknown code: ${approvalCode}`
-    };
-
-    return approvalText[approvalCode] || approvalText.default;
-};
+ * Returns a selector function which returns the unique classes/label/radius for the gw level points
+ * @return {Function} which returns an {Array of Object} with the following properties:
+ *      @prop {Array of String} classes
+ *      @prop {String} label
+ *      @prop {Number} radius
+ */
+export const getUniqueGWKinds = createSelector(
```

Comment:
There is an analogous function in iv-data which is used to determine what legend markers we need.

Reply:
Neat